presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -739,18 +743,78 @@ private ConnectorTableMetadata getTableMetadata(ConnectorSession session, Schema
     }
 
     @Override
-    public Optional<Object> getInfo(ConnectorTableLayoutHandle layoutHandle)
+    public Optional<Object> getInfo(ConnectorSession session, ConnectorTableLayoutHandle layoutHandle)
     {
         HiveTableLayoutHandle tableLayoutHandle = (HiveTableLayoutHandle) layoutHandle;
-        if (tableLayoutHandle.getPartitions().isPresent()) {
-            return Optional.of(new HiveInputInfo(
-                    tableLayoutHandle.getPartitions().get().stream()
-                            .map(HivePartition::getPartitionId)
-                            .collect(toList()),
-                    false));
+        if (!tableLayoutHandle.getPartitions().isPresent()
+                || 0 == tableLayoutHandle.getPartitions().get().size()) {
+            return Optional.empty();
         }
 
-        return Optional.empty();
+        MetastoreContext metastoreContext = new MetastoreContext(
+                session.getIdentity(),
+                session.getQueryId(),
+                session.getClientInfo(),
+                session.getSource(),
+                getMetastoreHeaders(session),
+                isUserDefinedTypeEncodingEnabled(session),
+                metastore.getColumnConverterProvider());
+
+        if (isUnpartitionedTable(tableLayoutHandle)) {
+            return getHiveInputInfoForUnpartitionedTable(metastoreContext, tableLayoutHandle);
+        }
+        return getHiveInputInfoForPartitionedTable(metastoreContext, tableLayoutHandle);
+    }
+
+    private boolean isUnpartitionedTable(HiveTableLayoutHandle tableLayoutHandle)
+    {
+        List<HivePartition> partitions = tableLayoutHandle.getPartitions().get();
+        if (1 == partitions.size()
+                && 0 == partitions.get(0).getPartitionId().compareTo(DUMMY_PARTITION_NAME)) {
+            return true;
+        }
+        return false;
+    }
+
+    private Optional<Object> getHiveInputInfoForUnpartitionedTable(
+            MetastoreContext metastoreContext, HiveTableLayoutHandle tableLayoutHandle)
+    {
+        Optional<Table> table = metastore.getTable(
+                metastoreContext,
+                tableLayoutHandle.getSchemaTableName().getSchemaName(),
+                tableLayoutHandle.getSchemaTableName().getTableName());
```

Comment:
These are very expensive metastore calls. Somehow I can't link this PR with previous PRs:
- Why do we need last access times in getInfo?
- What is the relationship between last commit time and last access time?

Reply:
Depa logging needs two kinds of information: the access time of the input (the input table/partition), and the data commit time of the output (the created table/partition). The commit time can only be obtained through the commit handle. The access time can be directly obtained by calling the metastore since data already exist. I am not sure if there are any cheaper ways to get these information.

Reply:
A bit more information: Depa wants to log the lastDataCommitTime. However, these open source table/partition APIs do not contain that field (I believe we should not take troubles to add it). Therefore, in presto-facebook, after we get the lastDataCommitTime from the metastore, and we will use the lastAccessTime field to store the lastDataCommitTime.

Reply:
Would be good if we have a meeting to discuss this in person. Somehow, I still can link the change of this PR to the previous ones. Would be nice if we can explain the end-to-end flow.

Also, we probably don't want to overload a field with too many meanings... There are other ways to store specialized fields.